src/main/java/org/owasp/benchmark/score/parsers/CheckmarxIASTReader.java

```diff
@@ -0,0 +1,162 @@
+        return 693; // Click_Jacking
+      case "Command_Injection":
+        return 78; // Command_Injection
+      case "CORS":
+        return 346; // CORS
+      case "CSRF":
+        return 352; // CSRF
+      case "Debug_Mode_Enabled":
+        return 215; // Debug_Mode_Enabled
+      case "Deserialize_Vulnerability":
+        return 502; // Deserialize_Vulnerability
+      case "Failed_Login_Without_Audit":
+        return 778; // Failed_Login_Without_Audit
+      case "File_Upload_To_Unprotected_Directory":
+        return 434; // File_Upload_To_Unprotected_Directory
+      case "Improper_HTTP_Get_Usage":
+        return 650; // Improper_HTTP_Get_Usage
+      case "Insecure_Cookie":
+      case "Session_Id_Disclosure": //CxIAST does not define but it is same as Insecure_Cookie YE
+        return 614; // Insecure_Cookie
+      case "Insecure_Outgoing_Communication":
+        return 311; // Insecure_Outgoing_Communication
+      case "Insufficient_Session_Expiration":
+        return 613; // Insufficient_Session_Expiration
+      case "LDAP_Injection":
+        return 90; // LDAP_Injection
+      case "Least_Privilege_Violation":
+        return 250; // Least_Privilege_Violation
+      case "Log_Forging":
+        return 117;
+      case "Missing_X_Content_Type_Options_Header":
+        return 693;
+      case "Missing_X_XSS_Protection_Header":
+        return 693;
+      case "NoSQL_Injection":
+        return 943;
+      case "Open_Redirect":
+        return 601;
+      case "Parameter_Pollution":
+        return 235;
+      case "Parameter_Tampering":
+        return 99;
+      case "Path_Traversal":
+        return 22;
+      case "Second_Order_Command_Injection":
+        return 77;
+      case "Second_Order_LDAP_Injection":
+        return 90;
+      case "Second_Order_Path_Traversal":
+        return 22;
+      case "Second_Order_SQL_Injection":
+        return 89;
+      case "Second_Order_XPath_Injection":
+        return 643;
+      case "Sensitive_Data_Exposure_Credit_Card":
+        return 311;
+      case "Sensitive_Data_Exposure_Email":
+        return 311;
+      case "Sensitive_Data_Exposure_Long_Number":
+        return 311;
+      case "SQL_Injection":
+        return 89;
+      case "Stored_XSS":
+        return 79;
+      case "Successful_Login_Without_Audit":
+        return 778;
+      case "Trust_Boundary_Violation":
+        return 501;
+      case "Weak_Cryptography":
+        return 327;
+      case "Weak_DB_Password":
+        return 521;
+      case "Weak_Hashing":
+        return 328;
+      case "Weak_Random":
+        return 330;
+      case "XPath_Injection":
+        return 643;
+      case "XSS":
+        return 79;
+      case "XXE":
+        return 611;
+    }
+    return 0;
+  }
+
+  public TestResults parse(File f) throws Exception
+  {
+    TestResults tr = new TestResults("CxIAST", true, TestResults.ToolType.IAST);
+
+    java.io.Reader inReader = new java.io.FileReader(f);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *inReader*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by CSVFormat.parse. Either a) close *inReader* in a try-finally block or b) close the resource by declaring *inReader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).